python/katana/lonestar/analytics/deg_count.py

```diff
@@ -0,0 +1,68 @@
+import numpy as np 
+import pyarrow
+
+from katana.property_graph import PropertyGraph
+from katana.datastructures import LargeArray, AllocationPolicy
+from katana.loops import do_all, do_all_operator
+from katana.atomic import (
+    atomic_add,
+)
+
+@do_all_operator()
+def initialize_in_deg(nin, nid):
+    nin[nid] = 0
+
+@do_all_operator()
+def count_in_and_out_deg(graph: PropertyGraph, nout, nin, nid):
+    outdeg = 0
+    for edge in graph.edges(nid):
+        outdeg += 1
+        dst = graph.get_edge_dst(edge)
+        atomic_add(nin, dst, 1)
+    nout[nid] = outdeg
+
+@do_all_operator()
+def count_weighted_in_and_out_deg(graph: PropertyGraph, nout, nin, weightArr, nid):
+    outdeg = 0
+    for edge in graph.edges(nid):
+        weight = weightArr[edge]
+        outdeg += weight
+        dst = graph.get_edge_dst(edge)
+        atomic_add(nin, dst, weight)
+    nout[nid] = outdeg
+
+""" 
+   Calculate the (potentially weighted) in & out degrees of a graph
+   Will add the in & out degrees as 2 new node properties of the graph
+   weight_property: an edge property to use in calculating the weighted degree
+   inProp: the property name for the in degree
+   outProp: the property name for the out degree
+"""
```

Comment:
Doc strings go inside the function. https://www.pythonforbeginners.com/basics/python-docstrings

Also please write in full sentences. The documentation will eventually show up to users.